config/config.go

```diff
@@ -32,7 +36,13 @@ func New() (*Config, error) {
 	}
 	flag.Parse()
 	err := config.loadConfigFromFile()
-	config.setLogLevel()
+
+	if err != nil {
+		return config, err
```

Comment:
I think we should return `nil, err` here as config is already defined as `Config{}`